src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -434,30 +1101,34 @@ func (s *commitLogSource) mergeSeries(
 	var numEmptyErrs int
 	var numErrs int
 
-	for startNano, encoders := range unmergedBlocks.encoders {
+	for startNano, encoders := range unmergedCommitlogBlocks.encoders {
 		start := startNano.ToTime()
 
-		if len(encoders) == 0 {
-			numEmptyErrs++
-			continue
+		var (
+			snapshotBlock    block.DatabaseBlock
+			hasSnapshotBlock bool
+		)
+
+		if snapshotData.Blocks != nil {
+			snapshotBlock, hasSnapshotBlock = snapshotData.Blocks.BlockAt(start)
 		}
 
-		if len(encoders) == 1 {
-			pooledBlock := blocksPool.Get()
-			pooledBlock.Reset(start, blockSize, encoders[0].enc.Discard())
-			if seriesBlocks == nil {
-				seriesBlocks = block.NewDatabaseSeriesBlocks(len(unmergedBlocks.encoders))
-			}
-			seriesBlocks.AddBlock(pooledBlock)
+		if !hasSnapshotBlock {
+			// Make sure snapshotBlock is nil if it does not exist.
+			snapshotBlock = nil
```

Comment:
Hm, huh?  Shouldn't `BlockAt(...)` guarentee `<notnil>, true` or `<nil>, false`?  And if not should we not just update it to guarentee that?

Reply:
Yeah that is the contract that function provides (and correctly implements). I was just being paranoid because newIOReadersFromEncodersAndBlock does a nil check that I didn't want to think about whether it would work properly if there was an empty block